src/main/java/org/csanchez/jenkins/plugins/kubernetes/ContainerTemplate.java

```diff
@@ -150,6 +151,17 @@ public void setEnvVars(List<ContainerEnvVar> envVars) {
         this.envVars.addAll(envVars);
     }
 
+    public List<PortMapping> getPorts() {
+        return ports != null ? ports : Collections.emptyList();
+    }
+
+    @DataBoundSetter
+    public void setPorts(List<PortMapping> ports) {
+        if (ports != null) {
+            this.ports.addAll(ports);
+        }
```

Comment:
@carlossg since the field is final you just can't set it like that.

I'm chaning it to this

```java
this.ports.clear();
this.ports.addAll(ports);
```

It would also seem that `setEnvVars` falls into the same problem, but that's another bug.